src/frontend/src/components/FillForms/UserDataForm.tsx

```diff
@@ -1,337 +1,294 @@
-import React from "react";
-import history from "../../service/history";
+import React, { useState } from "react";
 import { downloadExpungementPacket } from "../../redux/search/actions";
-import { connect } from "react-redux";
-import { AppState } from "../../redux/store";
-import { AliasData } from "../RecordSearch/SearchPanel/types";
 import InvalidInputs from "../InvalidInputs";
-import moment from "moment";
+import { useAppSelector } from "../../redux/hooks";
+import { useDispatch } from "react-redux";
+import EmptyFieldsModal from "./EmptyFieldsModal";
+import { selectSearchFormValues } from "../../redux/searchFormSlice";
+import { isBlank, objectToArray } from "../../service/validators";
+import { Redirect } from "react-router-dom";
 
-interface Props {
-  aliases: AliasData[];
-  loadingExpungementPacket: boolean;
-  downloadExpungementPacket: Function;
-}
+const isPresent = (str: string) => !isBlank(str);
 
-interface State {
-  name: string;
-  dob: string;
-  mailingAddress: string;
-  phoneNumber: string;
-  city: string;
-  state: string;
-  zipCode: string;
-  invalidZipCode: boolean;
-  invalidPhone: boolean;
-  invalidBirthDate: boolean;
-}
+const errorMessagesObjectToArray = (obj: object) => objectToArray(obj);
```

Comment:
This is more of a preference but `objectToArray` is currently only used once in this entire project in this file, so I would rather just keep the function in here. In the future, when it is used again, we can extract it as you did. It is a tough judgement call but I find that if I "prematurely" extract it whenever I can, it ends up never being used again half of the time, making it just harder to read/review half of the time. Of course if it is really generic then it makes sense but this `objectToArray` is only for specific types of objects.

Reply:
ok I changed it back